src/main/kotlin/no/nav/amt/deltaker/bff/kafka/ManagedKafkaConsumer.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.amt.deltaker.bff.kafka
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+class ManagedKafkaConsumer<K, V>(
+    val topic: String,
+    val config: Map<String, *>,
+    val consume: (key: K, value: V) -> Unit,
+) {
+    private val log = LoggerFactory.getLogger(javaClass)
+    private val job = Job()
+    private val scope = CoroutineScope(Dispatchers.Default + job)
```

Comment:
Jeg lurer på om vi kanskje egentlig skal bruke GlobalScope og ikke CoroutineScope her? Global scope vil vel gjøre at denne prosessen kjører for seg selv uten å bli avbrutt av andre ting så lenge appen kjører, mens coroutine scope er mer for kortlivede oppgaver som skal pauses og startes og avsluttes, hvis ikke jeg husker feil?

Reply:
Job er heller ikke noe jeg har brukt før. Tidligere har jeg brukt en egen dispatcher som ser sånn her ut og som egentlig er en ExecutorCoroutineDispatcher: 
```
class UnboundedDispatcher private constructor() : CoroutineDispatcher() {
    companion object {
        val unboundedDispatcher = UnboundedDispatcher()
    }

    private val threadPool = newCachedThreadPool()
    private val dispatcher = threadPool.asCoroutineDispatcher()
    override fun dispatch(context: CoroutineContext, block: Runnable) {
        dispatcher.dispatch(context, block)
    }
}
```